File: SYP.Web/Modules/Setting/Country/CountryForm.cs
namespace SYP.Setting.Forms;

[FormScript("Setting.Country")]
[BasedOnRow(typeof(CountryRow), CheckNames = true)]
public class CountryForm
{
    public string Name { get; set; }
    public string Code { get; set; }
    public string IsoCode3 { get; set; }
    public string PhoneCode { get; set; }
    public string CountryNr { get; set; }
    public int InsertUserId { get; set; }
    public DateTime InsertDate { get; set; }
    public int UpdateUserId { get; set; }
    public DateTime UpdateDate { get; set; }
}

File: SYP.Web/Modules/ServerTypes/Setting/CountryForm.ts
import { DateEditor, initFormType, IntegerEditor, PrefixedContext, StringEditor } from "@serenity-is/corelib";

export interface CountryForm {
    Name: StringEditor;
    Code: StringEditor;
    IsoCode3: StringEditor;
    PhoneCode: StringEditor;
    CountryNr: StringEditor;
    InsertUserId: IntegerEditor;
    InsertDate: DateEditor;
    UpdateUserId: IntegerEditor;
    UpdateDate: DateEditor;
}

export class CountryForm extends PrefixedContext {
    static readonly formKey = 'Setting.Country';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!CountryForm.init) {
            CountryForm.init = true;

            var w0 = StringEditor;
            var w1 = IntegerEditor;
            var w2 = DateEditor;

            initFormType(CountryForm, [
                'Name', w0,
                'Code', w0,
                'IsoCode3', w0,
                'PhoneCode', w0,
                'CountryNr', w0,
                'InsertUserId', w1,
                'InsertDate', w2,
                'UpdateUserId', w1,
                'UpdateDate', w2
            ]);
        }
    }
}